### DISCUSSION TITLE
Exception counts for thrown exception classes

### DISCUSSION THREAD

**Discussion by matthias-huber** [inspectIT/inspectit-ocelot#1385] [Q&A]
Is it possible with the standard instrumentation to track exceptions by class as metrics? I haven't found something in the documentation and in the default instrumentation.

Potentially a constructor instrumentation on Throwable should be the trick? Not sure how this would negatively affect application performance.

Any hints or opinions?

**Comment by mariusoe** [comment]
Hi @matthias-huber 

just to understand your goal: you want to know the amount of exception thrown? That should be possible by using a specific configuration. As you mentioned, you could instrument the exception constructor and count the amount objects created. But tracking the constructor invocation does not mean that the exception is actually be thrown.

The performance overhead should not be a problem, i would say.

**Comment by matthias-huber** [comment]
Hi @mariusoe,

yes, correct. At the end, I want to see the (thrown) exceptions. Similar to the [SQL Browser](https://inspectit-performance.atlassian.net/wiki/spaces/DOC18/pages/93009316/Working+with+the+SQL+Browser) from inspectIT.

As you said, it is probably not as easy as counting constructor calls. Probably, it should be done as 'derived' metric during tracing. It looks like that the [Exception Sensor](https://inspectit-performance.atlassian.net/wiki/spaces/DOC18/pages/93009346/Working+with+the+Exception+sensor) from inspectIT also worked on traces.
